validate.go

```diff
@@ -432,22 +433,37 @@ func (v *Validator) HexColor(key, value string, message ...string) {
 // Len sets the minimum and maximum length for a string.
 //
 // A maximum of 0 indicates there is no upper limit.
-func (v *Validator) Len(key, value string, min, max int, message ...string) {
+func (v *Validator) Len(key string, value interface{}, min, max int, message ...string) {
 	msg := getMessage(message, "")
-
-	switch {
-	case len(value) < min:
-		if msg != "" {
-			v.Append(key, msg)
-		} else {
-			v.Append(key, fmt.Sprintf(MessageLenLonger, min))
+	t := reflect.TypeOf(value)
+	k := t.Kind()
+	val := reflect.ValueOf(value)
```

Comment:
Why use relfect and not a much simpler type assertion?

Reply:
It's quite impossible to do type assertion for all types of maps, slices, etc, right?

Reply:
Do you need "all"? I think you probably need just `[]string`, and maybe `[]int`, and `[]int64`?

Reply:
Why wouldn't you support all with simpler/cleaner code?

Reply:
Reflection is known to be slow (hence so many 3rd party json libs try to avoid it)

Reply:
Performance should be fine for code like this. It's more about type safety and complexity; reflection is a lot more complex than type assertions.